### PR TITLE
Add support for Rx/Ry/Rz gates

### PR DESCRIPTION
**Pull request by BichengYing**
It is common to use Rx/Ry/Rz + some rotation degree. However, in current API, either there is known gates like `X,Y,Z,S,T,V`...., or you have to call customer gates like `customer,1,0,0,0.25,label`. 

We should be able to simply call the rotation like`animate_bloch rz 'Rz,-0.25'`.

### CODE REVIEW

File: bloch_sphere/animate_bloch.py [hunk 1]
```diff
@@ -142,6 +143,15 @@ def apply_gate_list(self, gates, final_wait=True):
                     print(f'Error: Custom gate arguments contain invalid '
                           f'floats {gate}.')
                 continue
+            if re.match('R[x,y,z][,;]', gate[:3]):
```
Comment (cduck) [inline bloch_sphere/animate_bloch.py]: Can you make it lowercase or case insensitive?

File: bloch_sphere/animate_bloch.py [hunk 2]
```diff
@@ -142,6 +143,15 @@ def apply_gate_list(self, gates, final_wait=True):
...
+                try:
+                    x, y, z = gate[1] == 'x', gate[1] == 'y', gate[1] == 'z' 
+                    r_pi = float(gate[3:])
+                    label = f"{gate[:2]}({r_pi:.3f}".rstrip('0').rstrip('.') + ")"
+                    self.custom_gate(x, y, z, r_pi, label=label)
+                except ValueError:
```
Comment (cduck) [inline bloch_sphere/animate_bloch.py:150]: Put the try/catch around the fewest possible lines to catch the float parsing error.  Either the r_pi line or the custom_gate line.

File: bloch_sphere/animate_bloch.py [hunk 3]
```diff
@@ -142,6 +143,15 @@ def apply_gate_list(self, gates, final_wait=True):
...
+                    print('Error: Rx/Ry/Rz gate should have style like Rx;{float} or Rx,{float}.')
```
Comment (cduck) [inline bloch_sphere/animate_bloch.py]: Wrap long lines to 80 characters if possible.

**Review by cduck — CHANGES_REQUESTED** [review]
Looks good.  Can you improve these couple things before I merge?

File: bloch_sphere/animate_bloch.py (hunk 1 above)
Comment (BichengYing) [reply bloch_sphere/animate_bloch.py]: Done

File: bloch_sphere/animate_bloch.py (hunk 3 above)
Comment (BichengYing) [reply bloch_sphere/animate_bloch.py]: Done.

File: bloch_sphere/animate_bloch.py (hunk 2 above)
Comment (BichengYing) [reply bloch_sphere/animate_bloch.py:150]: Done

File: bloch_sphere/animate_bloch.py [hunk 4]
```diff
@@ -142,6 +143,22 @@ def apply_gate_list(self, gates, final_wait=True):
...
+                    r_pi = float(gate[3:])
+                except ValueError:
+                    print('Error: Rx/Ry/Rz gate should have style like '
+                          'Rx;{float} or Rx,{float}.')
+                gate_name = gate[:3].lower()
+                x = int(gate_name[1] == 'x')
+                y = int(gate_name[1] == 'y')
+                z = int(gate_name[1] == 'z') 
+                formated_r_pi = f"{r_pi:.3f}".rstrip('0').rstrip('.')
+                label = f"{gate_name}({formated_r_pi})"
```
Comment (cduck) [inline bloch_sphere/animate_bloch.py]: Do you still want the label to be capitalized like Rx?

**Review by cduck — APPROVED** [review]

File: bloch_sphere/animate_bloch.py (hunk 4 above)
Comment (BichengYing) [reply bloch_sphere/animate_bloch.py]: Personally, I think Rx looks better than rx. This change will make it print 'rx'. If you don't object, I will update it to Rx instead.

Comment (BichengYing) [reply bloch_sphere/animate_bloch.py]: Done. Glad to check it again (The previous code had a mistake that printing message includes ;)

File: bloch_sphere/animate_bloch.py [hunk 5]
```diff
@@ -142,6 +143,22 @@ def apply_gate_list(self, gates, final_wait=True):
...
+            if re.match('r[x,y,z][,;]', gate[:3].lower()):
+                try:
+                    r_pi = float(gate[3:])
+                except ValueError:
+                    print('Error: Rx/Ry/Rz gate should have style like '
+                          'Rx;{float} or Rx,{float}.')
+                gate_name = 'R' + gate[1].lower()  # Display Rx instead of rx
+                x = int(gate_name[1] == 'x')
+                y = int(gate_name[1] == 'y')
+                z = int(gate_name[1] == 'z')
```
Comment (cduck) [inline bloch_sphere/animate_bloch.py]: A trailing space in here.

Comment (BichengYing) [reply bloch_sphere/animate_bloch.py]: Removed

File: bloch_sphere/animate_bloch.py [hunk 6]
```diff
@@ -142,6 +143,22 @@ def apply_gate_list(self, gates, final_wait=True):
...
+                try:
+                    r_pi = float(gate[3:])
+                except ValueError:
+                    print('Error: Rx/Ry/Rz gate should have style like '
+                          'Rx;{float} or Rx,{float}.')
+                gate_name = 'R' + gate[1].lower()  # Display Rx instead of rx
+                x = int(gate_name[1] == 'x')
+                y = int(gate_name[1] == 'y')
+                z = int(gate_name[1] == 'z') 
+                formated_r_pi = f"{r_pi:.3f}".rstrip('0').rstrip('.')
```
Comment (cduck) [inline bloch_sphere/animate_bloch.py:157]: Forgot one thing.  Add π after the number to match convention.

Comment (BichengYing) [reply bloch_sphere/animate_bloch.py:157]: Ah, Good point. Added

**Comment by cduck** [comment]
Published in [version 0.3.1](https://pypi.org/project/bloch-sphere/0.3.1/).

**Pull request merged**